src/axom/core/MapCollection.hpp

```diff
@@ -222,7 +220,7 @@ class MapCollection : public ItemCollection<T>
   const std::string& getItemName(IndexType idx) const
   {
     return (hasItem(idx) ? m_items[static_cast<unsigned>(idx)]->getName()
```

Comment:
@nselliott -- This function requires the templated type `T` to have a `getName()` function.  

Could you please document this requirement?
Alternatively, is there a way to conditionally include this function via `enable_if` ?

Reply:
This requirement is documented in the header.  I think it's better not to add a SFINAE implementation, as the core feature of `MapCollection` is to hold a collection of named items.  Using `enable_if` would make the code more complicated and allow instances of `MapCollection` to compile that don't have a meaningful use case, so in my opinion it's better to catch attempts to do that at compile time.